crates/bevy_asset/src/server/mod.rs

```diff
@@ -811,44 +825,57 @@ impl AssetServer {
             AssetServerMode::Processed { .. } => source.processed_reader()?,
         };
         let reader = asset_reader.read(asset_path.path()).await?;
-        match asset_reader.read_meta_bytes(asset_path.path()).await {
-            Ok(meta_bytes) => {
-                // TODO: this isn't fully minimal yet. we only need the loader
-                let minimal: AssetMetaMinimal = ron::de::from_bytes(&meta_bytes).map_err(|e| {
-                    AssetLoadError::DeserializeMeta {
-                        path: asset_path.clone_owned(),
-                        error: Box::new(DeserializeMetaError::DeserializeMinimal(e)),
-                    }
-                })?;
-                let loader_name = match minimal.asset {
-                    AssetActionMinimal::Load { loader } => loader,
-                    AssetActionMinimal::Process { .. } => {
-                        return Err(AssetLoadError::CannotLoadProcessedAsset {
-                            path: asset_path.clone_owned(),
-                        })
-                    }
-                    AssetActionMinimal::Ignore => {
-                        return Err(AssetLoadError::CannotLoadIgnoredAsset {
+        let read_meta = match &self.data.meta_mode {
+            AssetMetaMode::Always => true,
+            AssetMetaMode::Paths(paths) => paths.contains(asset_path),
```

Comment:
Requiring a `HashSet` containing all the exact paths of assets you want `.meta` files for feels incredibly cumbersome. 

It feels like this would want to support something like [`globset`](https://crates.io/crates/globset) or similar.  

An alternative would be to just expose this at the `AssetSource` level, and expose a new source `metaless://` (bikeshedding required). Then all the configuration for which Assets need `.meta` files moves to the asset use location instead of when the loader is configured. This reduces the chance of forgetting to add something to the set.

Reply:
Bike shedding my own comment if `AssetSources` could take parameters you could have `source+meta://` and `source-meta://` for explicit use and then allow global default configuration at the plugin level (Always vs Never mode)

Reply:
I almost didn't include the hashset at all. This is intended to be a "roughly non-breaking patch fix". The idea behind the hashset is not to be a perfect / recommended solution, but rather a simple short term "break glass" solution for cases that absolutely need it.

General theory is that at this point, most people are either using no meta files, or they are using asset processing (and therefore everything has meta files). The hashset covers the middle ground case (only needed on WASM) while we sort out the right long term fix.

Reply:
That seems reasonable, I somehow glossed over the "patch fix" information in the description. 

I'd be happy to circle back around on the topic if/when you look at `.meta` file configuration again in the future.